test/controllers/confirm.company.controller.unit.ts

```diff
@@ -0,0 +1,17 @@
+jest.mock("ioredis");
```

Comment:
should be able to remove the ioredis now as it will be imported from the session middleware mock (included in all.middleware.mocks)

Reply:
When removed (the test still passes, but) you get the warning
 `A worker process has failed to exit gracefully and has been force exited. This is likely caused by tests leaking due to improper teardown. Try running with --detectOpenHandles to find leaks.`

How about moving it to the middleware mock class?

Reply:
ok, you can leave it and i'll investigate. it should already be moved to the mock middleware, i can take a look after merged.